modules/Sources/Features/ImportWallet/ImportWallet.swift

```diff
@@ -60,31 +61,52 @@ public struct ImportWallet: Reducer {
     }
     
     public var body: some ReducerOf<ImportWallet> {
+        BindingReducer()
+        
         Reduce { state, action in
             switch action {
             case .alert(.dismiss):
                 return .none
-            case .birthdayInputChanged(let redactedBirthday):
+            case .binding(\.$birthdayHeight):
                 let saplingActivation = saplingActivationHeight
-
-                state.birthdayHeight = redactedBirthday
-
-                if let birthdayHeight = BlockHeight(state.birthdayHeight.data), birthdayHeight >= saplingActivation {
+                if let birthdayHeight = BlockHeight(state.birthdayHeight), birthdayHeight >= saplingActivation {
```

Comment:
This make me think that it's possible that you want `birthdayHeightValue` to be a computed property based on `state.birthdayHeight.data` instead

Reply:
Yeah, that might make more sense.